gcloud-java-compute/src/main/java/com/google/gcloud/compute/RegionAddressId.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Identity for a Google Compute Engine region address.
+ */
+public final class RegionAddressId extends RegionResourceId implements AddressId {
```

Comment:
Please have a look at the operation hierarchy that has already gone through review, this is very similar. Making `RegionAddressId` extend `GlobalAddressId` may help reusing code (not so much in fact) but makes no sense from a conceptual standpoint.

To avoid code duplication, `RegionResourceId` and `ZoneResourceId` are already being used.